warehouse/admin/templates/admin/projects/detail.html

```diff
@@ -132,4 +145,30 @@ <h3 class="box-title">Blacklist Project</h3>
       </div>
     </form>
   </div>
+
+  <!-- Upload limit form -->
+  <div class="box box-secondary collapsed-box">
+    <div class="box-header with-border">
+      <h3 class="box-title">Set upload limit</h3>
+      <div class="box-tools">
+        <button class="btn btn-box-tool" data-widget="collapse" data-toggle="tooltip" title="Expand"><i class="fa fa-plus"></i></button>
+      </div>
+    </div>
+
+    <form method="POST" action="{{ request.route_path('admin.project.set_upload_limit', project_name=project.normalized_name) }}">
+      <input name="csrf_token" type="hidden" value="{{ request.session.get_csrf_token() }}">
+      <div class="box-body">
+        <div class="form-group col-sm-12">
+          <label for="uploadLimit">Upload limit (in bytes)</label>
+          <input type="text" name="upload_limit" class="form-control" id="uploadLimit" rows="3" value="{{ project.upload_limit | default('', True)}}">
```

Comment:
This would be a bit nicer as ``<input type="number" min={{ THE MINMUM SIZE }} ...>`` instead of as a text field. It might also make sense to include ``step="10"`` or something, which i *think* means that the arrows that browsers render this with will increment/decrement this by 10, but you can still type in whatever step you want directly into the box.